WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -563,9 +565,38 @@ private static async Task WaitAndHandleResultOfTasksAsync(string logPrefix, para
 		}
 	}
 
-	public void WalletEnteredSendWorkflow(string walletName) => WalletsInSendWorkflow.TryAdd(walletName, 0);
+	public void WalletEnteredSendWorkflow(string walletName)
+	{
+		if (CoinJoinClientStates.TryGetValue(walletName, out var stateHolder) && 
+		    stateHolder.CoinJoinClientState is not CoinJoinClientState.Idle &&
+		    stateHolder.CoinJoinClientState is not CoinJoinClientState.InCriticalPhase && 
+		    !WalletsStoppedBecauseSendWorkflow.Contains(walletName))
```

Comment:
My problem with this is if the wallet enters the `SendWorkFlow` while the CJ is in `Critical Phase`, we will never add the `walletName` to the `WalletsStoppedBecauseSendWorkflow`. Thus we will never hit this sanity check

```cs
if (WalletsStoppedBecauseSendWorkflow.Contains(walletToStart.WalletName))
{
	throw new CoinJoinClientException(CoinjoinError.UserInSendWorkflow);
}
```
and we will never stop CJing, even if the wallet is in the `TxPreviewDialog/SendWorkflow`.

I tried to simplify it, but the restarting mechanism in `WalletLeftSendWorkflow()` became problematic.

IMO using the `WalletStatus` as in my original PR is still the easiest way to do this. 
Plus if we need to add further functionality for similar scenarios my PR is easier to scale IMO.

Reply:
Thanks @turbolay, I simplified the logic in my PR. 
Although, I modified your code a bit.

Please take a look if you have some spare time.

https://github.com/zkSNACKs/WalletWasabi/pull/10500/commits/c8545fa08933dd1ee167b684d46a4632114976e9